### ISSUE TITLE
[no-unnecessary-type-assertion] support const assertions

### ISSUE THREAD

**Issue by xiaoxiangmoe** [typescript-eslint/typescript-eslint#1734]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unnecessary-type-assertion": ["error"]
  }
}
```

```TS
const a = [1, 2];
const b = [3, 4];
const c = [...a, ...b] as const;  // This assertion is unnecessary since it does not change the type of the expression.eslint(@typescript-eslint/no-unnecessary-type-assertion)
```

**Expected Result**
No Error
**Actual Result**
See Error
**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.23.0` |
| `@typescript-eslint/parser`        | `2.23.0` |
| `TypeScript`                       | `3.7.5` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `13.5.0` |
| `npm`                              | `6.13.4` |

**Comment by bradzacher** [comment]
shouldn't be all that hard to add support to the rule for the `const` keyword.
happy to accept a PR
